src/h5cpp/datatype/datatype.cpp

```diff
@@ -47,24 +48,35 @@ Datatype::Datatype(ObjectHandle &&handle)
 }
 
 Datatype::Datatype(const Datatype &type) {
-  hid_t ret = H5Tcopy(static_cast<hid_t>(type.handle_));
-  if (0 > ret) {
-    error::Singleton::instance().throw_with_stack("Could not copy-construct Datatype");
+  if(static_cast<hid_t>(type.handle_)) {
```

Comment:
maybe it's better to create a method for those lines , same code is used again later

Reply:
I've just added a swap method with the common implementation